### PR TITLE
added convenience methods to create and use a collection when not exists

### PR DESCRIPTION
**Pull request by StephenMiracle**
I needed to add a couple of convenience methods so that I can create and use a collection in a single call.

### CODE REVIEW

**Comment by HouzuoGuo** [comment]
Many thanks, could you please put an RLock on schemaLock in ColExists as well?

File: db/db.go [hunk 1]
```diff
@@ -133,6 +133,30 @@ func (db *DB) Use(name string) *Col {
...
+	}
+
+	return db.Use(name)
+}
+
+
+// Check if collection exists
+func (db *DB) ColExists(name string) bool {
+	db.schemaLock.RLock()
+	defer db.schemaLock.RUnlock()
```
Comment (StephenMiracle) [inline db/db.go]: Not sure if this was necessary, but tried to follow same pattern you had elsewhere

File: db/db.go [hunk 2]
```diff
@@ -133,6 +133,30 @@ func (db *DB) Use(name string) *Col {
 	return nil
 }
 
+
+// Creates a collection if one doesn't exist and returns its use.
+func (db *DB) ForceUse(name string) *Col {
```
Comment (StephenMiracle) [inline db/db.go]: What do you think of this name? Had trouble thinking of better.

**Comment by StephenMiracle** [comment]
I'm getting an issue with a never loading page when trying to use the FOrceUse method and not quite sure why. Anything seem off to you?

**Comment by HouzuoGuo** [comment]
Oops, i think the functions are making too many attempts at acquiring schemaLock. It is indeed important to hold the lock while manipulating db.cols, don't worry, if there's a difficulty with it i'll take a look later this week.

**Comment by HouzuoGuo** [comment]
Here they are: https://github.com/HouzuoGuo/tiedot/commit/fda44628e562fb976eac72afc0231967fba817d2

**Comment by StephenMiracle** [comment]
awesome thanks!

**Comment by HouzuoGuo** [comment]
done over here: https://github.com/HouzuoGuo/tiedot/commit/fda44628e562fb976eac72afc0231967fba817d2

**Pull request closed without merging**